### ISSUE TITLE
Supporting AMD GPUs

### ISSUE THREAD

**Issue by samhatfield** [ecmwf-ifs/ectrans#125]
The situation on AMD GPUs for develop is a mess. Here are a list of crashes I am seeing on LUMI-G (MI250x) when running the benchmark driver (only `-t 79 --norms` passed):
- You usually have to build with a single thread because of `Error generating accelerator code: offload bundling failure`.
- 1 task/GPU: `ACC: Host object 0x2688380 does not appear to have an associated accelerator object.` Seems to occur inside [this call](https://github.com/ecmwf-ifs/ectrans/blob/main/src/trans/gpu/internal/trltog_mod.F90#L893). 
- \>= 2 tasks/GPUs (_sometimes_): same error as above.
- \>= 2 tasks/GPUs (_sometimes_): `HIPFFT_PARSE_ERROR`. Our old friend. For some reason we cannot plan FFTs, because of this cryptic error message.
- \>= 2 tasks/GPUs (_rarely_): NaNs in error norms. Otherwise completes.

CCE/16.0.1, ROCm/5.2.3.

So to sum up, we have two error messages with no discernible causes that don't always occur, and when they don't occur we get NaNs.

**Comment by samhatfield** [comment]
Latest update, based on CCE/17.0.1, ROCm/6.0.3, using benchmark command
```
--norms --truncation 79 --niter 100 --nlev 137 --nfld 1 --vordiv --uvders --scders --meminfo -v
```
:

- Still cannot build with more than 1 thread due to an apparent race condition.
- 1 task/GPU: there is a problem with in-place FFTs. I ran it 10 times and every time it crashed with `HIPFFT_EXEC_FAILED`. I have added an option to build without in-place FFTs, and enabled this option always for HIP. See PR #155. I think we can safely conclude that single GPU running is stable now.
- 2 tasks/GPUs: `Memory access fault by GPU node-4 (Agent handle: 0xa220aa0) on address 0x153d9560b000. Reason: Unknown.` I have a feeling that the non-GPU-aware code path has a bug in it.

For all of these I have disabled GPU-aware MPI, and graphs, to simplify debugging.

**Comment by samhatfield** [comment]
Latest release 1.5.0 has mostly stable performance on AMD GPUs. I'm going to close this issue.
